x/sequencer/keeper/msg_server.go

```diff
@@ -0,0 +1,58 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+
+	"github.com/decentrio/rollkit-sdk/x/sequencer/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+var _ types.MsgServer = msgServer{}
+
+// NewMsgServerImpl returns an implementation of the bank MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+func (k msgServer) ChangeSequencers(goCtx context.Context, msg *types.MsgChangeSequencers) (*types.MsgChangeSequencersResponse, error) {
+	if k.authority != msg.Authority {
+		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", k.authority, msg.Authority)
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if len(msg.Sequencers) != 0 {
+		newSequencer := msg.Sequencers[0]
+		err := k.Sequencer.Set(ctx, newSequencer)
+		if err != nil {
+			return nil, err
+		}
+		err = k.NextSequencerChangeHeight.Set(ctx, ctx.BlockHeight())
```

Comment:
blockHeigh should be set by `MsgChangeSequencers` instead of set by current block